tfep/io/tmbarlog.py

```diff
@@ -0,0 +1,718 @@
+
+    Current database format
+    -----------------------
+
+    The data is stored in HDF5 format with one file per epoch/step, containing groups
+    for different state mappings. The data is organized in the following structure:
+
+    ```
+    tfep_logs/
+    ├── train/
+    │   ├── epoch-0.h5
+    │   │   ├── 0_1/  # Mapping from state 0 to state 1
+    │   │   │   ├── potential_energy
+    │   │   │   ├── cv_values
+    │   │   │   └── ...
+    │   │   ├── 0_2/  # Mapping from state 0 to state 2
+    │   │   └── ...
+    │   ├── epoch-1.h5
+    │   └── ...
+    ├── eval/
+    │   ├── step-0.h5
+    │   │   ├── 0_1/
+    │   │   ├── 0_2/
+    │   │   └── ...
+    │   └── ...
+    └── metadata.json
+    ```
+
+    Each HDF5 file contains groups for each state mapping, with datasets for the
+    various quantities. The data is organized hierarchically:
+    - First level: state mapping groups (e.g., "0_1" for mapping from state 0 to state 1)
+    - Second level: datasets (potential_energy, cv_values, etc.)
+
+    Finally, a JSON file is used to store metadata about the experiment such as
+    batch and epoch sizes.
+
+    """
+
+    VERSION = '0.1'
+    METADATA_FILE_NAME = 'metadata.json'
+    INDEX_NAMES = ['trajectory_sample_index', 'dataset_sample_index']
+    MASK_NAME = '__mask'
+
+    def __init__(
+            self,
+            save_dir_path='tfep_logs',
+            data_loader=None,
+            train_subdir_name='train',
+            eval_subdir_name='eval',
+            n_states=2,
+            state_names: Optional[List[str]] = None,
+    ):
+        """Constructor.
+
+        Parameters
+        ----------
+        save_dir_path : str, optional
+            The main directory where to save the training and evaluation data.
+        data_loader : torch.utils.data.DataLoader, optional
+            The data loader used for training wrapping a :class:``tfep.io.dataset.TrajectoryDataset``.
+            This must be passed when a new logger is created as it is used to
+            determine epoch, batch, and trajectory dimensions. If ``save_dir_path``
+            points to an existing logger, then this is ignored.
+        train_subdir_name : str, optional
+            The name of the subdirectory where the training data is stored.
+        eval_subdir_name : str, optional
+            The name of the subdirectory where the evaluation data is stored.
+        n_states : int, optional
+            The number of states in the system. Default is 2.
+        state_names : List[str], optional
+            Names for each state. If None, defaults to ['state_0', 'state_1', ...].
+
+        """
+        self._save_dir_path = os.path.realpath(save_dir_path)
+        self._train_dir_path = os.path.join(save_dir_path, train_subdir_name)
+        self._eval_dir_path = os.path.join(save_dir_path, eval_subdir_name)
+        
+        # Initialize state parameters
+        self._n_states = n_states
```

Comment:
Can we maybe implement this as a property that returns `len(self._state_names)` to avoid state duplication?